tfhe/src/integer/gpu/ciphertext/compressed_ciphertext_list.rs

```diff
@@ -40,11 +53,133 @@ impl CudaCompressedCiphertextList {
             streams,
         )
     }
+
+    pub fn to_compressed_ciphertext_list(&self, streams: &CudaStreams) -> CompressedCiphertextList {
+        let glwe_list = self
+            .packed_list
+            .glwe_ciphertext_list
+            .to_glwe_ciphertext_list(streams);
+        let ciphertext_modulus = self.packed_list.glwe_ciphertext_list.ciphertext_modulus();
+
+        let first_element = self.packed_list.block_info.first().unwrap();
+        let message_modulus = first_element.message_modulus;
+        let carry_modulus = first_element.carry_modulus;
+        let pbs_order = first_element.pbs_order;
+        let lwe_per_glwe = self.packed_list.lwe_per_glwe;
+        let storage_log_modulus = self.packed_list.storage_log_modulus;
+
+        let initial_len = self.packed_list.initial_len;
+        let number_bits_to_pack = initial_len * storage_log_modulus.0;
+        let len = number_bits_to_pack.div_ceil(u64::BITS as usize);
+
+        let modulus_switched_glwe_ciphertext_list = glwe_list
+            .iter()
+            .map(|x| {
+                let glwe_dimension = x.glwe_size().to_glwe_dimension();
+                let polynomial_size = x.polynomial_size();
+                CompressedModulusSwitchedGlweCiphertext {
+                    packed_integers: PackedIntegers {
+                        packed_coeffs: x.into_container()[0..len].to_vec(),
+                        log_modulus: storage_log_modulus,
+                        initial_len,
+                    },
+                    glwe_dimension,
+                    polynomial_size,
+                    bodies_count: LweCiphertextCount(self.packed_list.bodies_count),
+                    uncompressed_ciphertext_modulus: ciphertext_modulus,
+                }
+            })
+            .collect_vec();
+
+        let count = CiphertextCount(self.packed_list.bodies_count);
+        let packed_list = ShortintCompressedCiphertextList {
+            modulus_switched_glwe_ciphertext_list,
+            ciphertext_modulus,
+            message_modulus,
+            carry_modulus,
+            pbs_order,
+            lwe_per_glwe,
+            count,
+        };
+
+        CompressedCiphertextList {
+            packed_list,
+            info: self.info.clone(),
+        }
+    }
+}
+
+impl CompressedCiphertextList {
+    pub fn to_cuda_compressed_ciphertext_list(
+        &self,
+        streams: &CudaStreams,
+    ) -> CudaCompressedCiphertextList {
+        let lwe_per_glwe = self.packed_list.lwe_per_glwe;
+
+        let modulus_switched_glwe_ciphertext_list =
+            &self.packed_list.modulus_switched_glwe_ciphertext_list;
+
+        let first_ct = modulus_switched_glwe_ciphertext_list.first().unwrap();
+        let storage_log_modulus = first_ct.packed_integers.log_modulus;
+        let initial_len = first_ct.packed_integers.initial_len;
+        let bodies_count = first_ct.bodies_count.0;
+
+        /// To-do: is there a better way to calculate degree?
```

Comment:
At this point I need to compute the expected Degree of each ciphertext. I couldn't find a good way to do it, so I'm calculating a whole LUT to get the max value and thus the degree (assuming the ct is fresh). @tmontaigu is there a better way?